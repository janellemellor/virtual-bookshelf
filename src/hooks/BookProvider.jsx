import React, { createContext, useState, useContext } from "react";
import { addBook } from '../services/booksApi';
 
const BookContext = createContext();

export const BookProvider = ({ children }) => {
  const [title, setTitle] = useState('');
  const [author, setAuthor] = useState('');
  const [genre, setGenre] = useState('');
  const [image, setImage] = useState('');
  const [synopsis, setSynopsis] = useState('');

  const onChange = ({ target }) => {
    const setFormInputsFactory = {
      title: setTitle,
      author: setAuthor,
      genre: setGenre,
      image: setImage, 
      synopsis: setSynopsis
    };

    setFormInputsFactory[target.name](target.value);
  };
  
  const onSubmit = (event) => {
      event.preventDefault();
      addBook({ title, author, genre, image, synopsis })
        .then(book => {

        });
  };

  const context = {
     onSubmit, 
     onChange, 
     title, 
     author, 
     genre, 
     image, 
     synopsis
  };

  return (
    <BookContext.Provider value={context} >
        {children}
    </BookContext.Provider>
  );
};

export const useBookProvider = () => {
    const context = useContext(BookContext);
    return context;
};



