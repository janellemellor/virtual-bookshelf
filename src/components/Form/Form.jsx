import React from 'react';
import { useBookProvider } from '../../hooks/BookProvider';

const Form = () => {
    const { onSubmit, onChange, title, author, genre, imageUrl, synopsis  } = useBookProvider();
    
  return (
    <form onSubmit={onSubmit}>
      <fieldset>
        <input name="title" type="text" value={title} onChange={onChange} placeholder="title" />
        <input name="author" type="text" value={author} onChange={onChange} placeholder="author" /> 
        <input name="genre" type="text" value={genre} onChange={onChange} placeholder="genre" />
        <input name="image" type="text" value={imageUrl} onChange={onChange} placeholder="image url" />
        <textarea name="synopsis" value={synopsis} onChange={onChange} placeholder="synopsis" >
        </textarea>
        <button>Add Book!</button>
      </fieldset>
    </form>    
  );  
};

export default Form;
