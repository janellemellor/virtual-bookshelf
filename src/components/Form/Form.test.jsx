import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';
import { BookProvider } from '../../hooks/BookProvider';

describe('Form Component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <BookProvider>
        <Form onSubmit={() => {}} onChange={() => {}} title='Salt' author='Nayyirah Waheed' genre='poetry' imageUrl='https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1380105266l/18585282.jpg' synopsis='It was only and ever love' />
      </BookProvider>);
      expect(wrapper).toMatchSnapshot();
  });  
});
