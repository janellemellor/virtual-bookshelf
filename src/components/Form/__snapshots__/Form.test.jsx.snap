// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component matches a snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "author": "",
      "genre": "",
      "image": "",
      "onChange": [Function],
      "onSubmit": [Function],
      "synopsis": "",
      "title": "",
    }
  }
>
  <Form
    author="Nayyirah Waheed"
    genre="poetry"
    imageUrl="https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1380105266l/18585282.jpg"
    onChange={[Function]}
    onSubmit={[Function]}
    synopsis="It was only and ever love"
    title="Salt"
  />
</ContextProvider>
`;
